src/main/java/jenkins/plugins/logstash/persistence/ElasticSearchDao.java

```diff
@@ -149,12 +159,27 @@ public String getMimeType() {
   public void setMimeType(String mimeType) {
     this.mimeType = mimeType;
   }
-  
+
+  public KeyStore getCustomKeyStore() {
+    return this.customKeyStore;
+  }
+
   String getAuth()
   {
     return auth;
   }
 
+  public void setCustomKeyStore(KeyStore customKeyStore) {
+    this.customKeyStore = customKeyStore;
+
+    try {
+      SSLHelper.setClientBuilderSSLContext(this.clientBuilder, this.customKeyStore);
+    } catch (KeyStoreException | CertificateException | NoSuchAlgorithmException
+            | KeyManagementException | IOException e) {
+      LOGGER.log(Level.WARNING, e.getMessage(), e);
```

Comment:
I think this can result in user trying to set the custom cert and getting no error messages in the UI while it had indeed failed. Can't we bubble this exception up to be shown immediately?

Reply:
I'm not sure how that would be accomplished. One way to do it might be to put in a 'test connection' button (irrespective of custom SSL cert or not),  but I feel that would be a separate issue and thereby would lie outside the scope of this PR.

Reply:
OK, I see this method isn't going to be called when you save the settings, only during the build.

Still, can you please change the following:
 - make `setCustomKeyStore` throw the exception and handle it (the same way) in calling method `createIndexerInstance()`
 - call `SSLHelper.setClientBuilderSSLContext` first and only then update the field (just to keep the internal state consistent)

Reply:
Resolved